misc/release_test.sh

```diff
@@ -0,0 +1,244 @@
+    local REPO="taichi"
+
+    # divider
+    taichi::utils::line
+    taichi::utils::logger::info "Running GGUI examples"
+    
+    # clone the repo
+    taichi::utils::git_clone "${ORG}" "${REPO}"
+    cd "${REPO}/python/taichi/examples/ggui_examples"
+
+    # run tests
+    for match in $(find ./ -name "${PATTERN}"); do
+        python "${match}"
+        taichi::utils::line
+        taichi::utils::pause
+    done
+
+    # go back to workdir
+    cd "${WORKDIR}"
+}
+
+function taichi::test::difftaichi {
+    local WORKDIR=${1}
+    local PATTERN="*.py"
+    local ORG="taichi-dev"
+    local REPO="difftaichi"
+
+    # divider
+    taichi::utils::line
+    taichi::utils::logger::info "Running DiffTaichi examples"
+    
+    # clone the repo
+    taichi::utils::git_clone "${ORG}" "${REPO}"
+    cd "${REPO}/examples"
+
+    # run tests
+    for match in $(find ./ -name "${PATTERN}"); do
+        python "${match}"
+        taichi::utils::line
+        taichi::utils::pause
+    done
+
+    # go back to workdir
+    cd "${WORKDIR}"
+}
+
+function taichi::test::taichi_elements {
+    local WORKDIR=${1}
+    local PATTERN="demo_*.py"
+    local ORG="taichi-dev"
+    local REPO="taichi_elements"
+
+    # divider
+    taichi::utils::line
+    taichi::utils::logger::info "Running Taichi Elements examples"
+    
+    # clone the repo
+    taichi::utils::git_clone "${ORG}" "${REPO}"
+    cd "${REPO}"
+
+    # install dependencies
+    python "download_ply.py"
+
+    # run tests
+    cd "${REPO}/demo" 
+    for match in $(find ./ -name "${PATTERN}"); do
+        python "${match}"
+        taichi::utils::line
+        taichi::utils::pause
+    done
+
+    # run special tests
+    python3 render_particles.py -i ./path/to/particles \
+                                -b 0 -e 400 -s 1 \
+                                -o ./output \
+                                --gpu-memory 20 \
+                                -M 460 \
+                                --shutter-time 0.0 \
+                                -r 128
+
+    # go back to workdir
+    cd "${WORKDIR}"
+}
+
+function taichi::test::stannum {
+    local WORKDIR=${1}
+    local ORG="ifsheldon"
+    local REPO="stannum"
+
+    # divider
+    taichi::utils::line
+    taichi::utils::logger::info "Running Stannum examples"
+    
+    # clone the repo
+    taichi::utils::git_clone "${ORG}" "${REPO}"
+    cd "${REPO}"
+
+    # run tests
+    pytest -v -s ./
+
+    # go back to workdir
+    cd "${WORKDIR}"
+}
+
+function taichi::test::sandyfluid {
+    local WORKDIR=${1}
+    local ORG="ethz-pbs21"
+    local REPO="SandyFluid"
+
+    # divider
+    taichi::utils::line
+    taichi::utils::logger::info "Running SandyFluid examples"
+    
+    # clone the repo
+    taichi::utils::git_clone "${ORG}" "${REPO}"
+    cd "${REPO}"
+
+    # install dependencies
+    # remove the line contains pinned Taichi version for testing purposes
+    grep -v "taichi" requirements.txt > tmpfile && mv tmpfile requirements.txt
```

Comment:
we probably need to be extra careful here as some people might hardcode taichi to a version in the requirements.txt so we may not be testing the new release :P

Reply:
Yup! That's what this line does, remove the line of `taichi==x.y.z` from `requirements.txt` before installing the dependencies. Do you think it's enough?

Reply:
> (btw we also need to run misc/generate_example_videos.py in the release test, shall we include it in the script as well?

we should... I had some issues with running that script while writing this one, e.g. I had to use `pytest` for some of the examples referenced by the script whereas run others within the script. For now I will leave that part for future work if you don't mind :P

Reply:
Aha sorry this looks good! Thanks!